apps/src/templates/rubrics/RubricFloatingActionButton.jsx

```diff
@@ -19,14 +19,42 @@ export default function RubricFloatingActionButton({
 }) {
   const [isOpen, setIsOpen] = useState(!!studentLevelInfo);
 
+  const eventData = useMemo(() => {
+    return {
+      ...reportingData,
+      viewingStudentWork: !!studentLevelInfo,
+      viewingEvaluationLevel: rubric.level.name === currentLevelName,
+    };
+  }, [reportingData, studentLevelInfo, rubric.level.name, currentLevelName]);
+
   const handleClick = () => {
     const eventName = isOpen
-      ? EVENTS.RUBRIC_CLOSED_FROM_FAB_EVENT
-      : EVENTS.RUBRIC_OPENED_FROM_FAB_EVENT;
-    analyticsReporter.sendEvent(eventName, reportingData);
+      ? EVENTS.TA_RUBRIC_CLOSED_FROM_FAB_EVENT
+      : EVENTS.TA_RUBRIC_OPENED_FROM_FAB_EVENT;
+    analyticsReporter.sendEvent(eventName, eventData);
     setIsOpen(!isOpen);
   };
 
+  useEffect(() => {
+    if (!!studentLevelInfo) {
+      analyticsReporter.sendEvent(
+        EVENTS.TA_RUBRIC_ON_STUDENT_WORK_LOADED,
+        eventData
+      );
+    }
+
+    const fireUnloadEvent = () =>
+      analyticsReporter.sendEvent(
+        EVENTS.TA_RUBRIC_ON_STUDENT_WORK_UNLOADED,
+        eventData
+      );
+    window.addEventListener('beforeunload', fireUnloadEvent);
+
+    return () => {
+      window.removeEventListener('beforeunload', fireUnloadEvent);
+    };
+  }, [eventData, studentLevelInfo]); // Neither of these should change, so this should run once
```

Comment:
I am a bit torn here because I do want this to fire exactly once but the linter insists on listing dependencies. In this case, the dependencies shouldn't change (even including the `eventData` dependencies) but it still makes me a little nervous.

Reply:
this SO answer validates your complaint and talks about useMount: 
* https://stackoverflow.com/questions/53120972/how-to-call-loading-function-with-react-useeffect-only-once
* https://hooks-guide.netlify.app/react-use/useMount

If you don't want to invest in a new hook, I would suggest disabling the warning (with an explanation) over implementing uncertain/undesirable behavior just to satisfy lint.

Reply:
there's also an outside chance that inlining the very simple implementation will be enough to trick the linter:
```
const useMount = mount => useEffect(mount, []);
```

Reply:
I went down a rabbit hole and read a discussion of this linter rule and was more convinced as to why exhaustive dependencies make sense, particularly [this comment](https://github.com/facebook/react/issues/14920#issuecomment-470945051). I started to think about all the things that could make this effect rerun and they are:
- the student info changes
- reporting data changes -- that currently includes level name, unit name, and course name
- currentLevelName changes
- the rubric level changes

Condensing that list -- the only way this would rerun is if the student or level changed. In this case, I think we might actually want to rerun this useEffect. I was worried about stacking event listeners but learned this morning that the useEffect cleanup happens before executing the next effect ([React documentation link](https://react.dev/reference/react/useEffect#parameters)). So, if we were in a world where we changed level or student without page reloads, I think this would actually work as intended.